src/Components/Aspire.Microsoft.Azure.Cosmos/AspireMicrosoftAzureCosmosExtensions.cs

```diff
@@ -36,6 +36,42 @@ public static void AddAzureCosmosClient(
         AddAzureCosmosClient(builder, configureSettings, configureClientOptions, connectionName, serviceKey: null);
     }
 
+    /// <summary>
+    /// Registers <see cref="CosmosClient" /> and <see cref="Database"/> as singletons in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">The connection name to use to find a connection string.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="MicrosoftAzureCosmosSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientOptions">An optional method that can be used for customizing the <see cref="CosmosClientOptions"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire:Microsoft:Azure:Cosmos" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddAzureCosmosDatabase(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<MicrosoftAzureCosmosSettings>? configureSettings = null,
+        Action<CosmosClientOptions>? configureClientOptions = null)
+    {
+        AddAzureCosmosClient(builder, configureSettings, configureClientOptions, connectionName, serviceKey: null);
```

Comment:
How does this know to add the `Database` as a DI service? What's the difference between this and `AddAzureCosmosClient`?

From looking at the code, it appears to be based on the connection string and not which API was called.

Reply:
> From looking at the code, it appears to be based on the connection string and not which API was called.

Correct -- we can probably shift the parsing of the connection string earlier and throw if the string associated with `connectionName` doesn't contain all the parts associated with the resource type you're trying to add.

Reply:
Per offline discussion, we no longer register the client if you call the specific `AddAzureCosmosDatabase` or `AddAzureCosmosContainer` method.